package/kedro_viz/server.py

```diff
@@ -474,7 +477,61 @@ def format_pipeline_data(
 @app.route("/api/nodes.json")
 def nodes_json():
     """Serve the pipeline data."""
-    return jsonify(data)
+    return jsonify(_DATA)
+
+
+@app.route("/api/nodes/<string:node_id>")
+def nodes_metadata(node_id):
+    """Serve the metadata for node and dataset."""
+    if _PIPELINES:
+        for pipeline in _PIPELINES.values():
+            for node in sorted(pipeline.nodes, key=lambda n: n.name):
+                task_id = _hash(str(node))
```

Comment:
In `format_pipeline_data`, can I suggest we keep a global dictionary of `node_id -> node` so we don't have to keep searching for the node every time by looping through all pipelines? That way we also don't have to perform extra logic to match dataset with node id.